From 900fc60b3d56661b0f8e28acc605b808418046f9 Mon Sep 17 00:00:00 2001
From: Laszlo Agocs <lagocs83@gmail.com>
Date: Fri, 10 Oct 2025 10:26:08 +0200
Subject: [PATCH 213/239] rendercontrol test: Skip when format is not supported

Pick-to: 6.9 6.8
Fixes: QTBUG-141060
Change-Id: Ia57d64905e327ff03ae8bb223603e5a289c6342b
Reviewed-by: Shawn Rutledge <shawn.rutledge@qt.io>
(cherry picked from commit 0ef12acbaa94780e2c09995ddeb1aba807a49dd0)
Reviewed-by: Qt Cherry-pick Bot <cherrypick_bot@qt-project.org>
---
 .../auto/quick/qquickrendercontrol/tst_qquickrendercontrol.cpp | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/tests/auto/quick/qquickrendercontrol/tst_qquickrendercontrol.cpp b/tests/auto/quick/qquickrendercontrol/tst_qquickrendercontrol.cpp
index 08e118de7c..124e22c8fe 100644
--- a/tests/auto/quick/qquickrendercontrol/tst_qquickrendercontrol.cpp
+++ b/tests/auto/quick/qquickrendercontrol/tst_qquickrendercontrol.cpp
@@ -736,6 +736,9 @@ void tst_RenderControl::renderAndReadBackWithVulkanAndCustomDepthTexture()
     QRhi *rhi = renderControl->rhi();
     Q_ASSERT(rhi);
 
+    if (!rhi->isTextureFormatSupported(QRhiTexture::D24S8))
+        QSKIP("24-bit depth format not supported, skipping test");
+
     const QSize size = rootItem->size().toSize();
     QScopedPointer<QRhiTexture> tex(rhi->newTexture(QRhiTexture::RGBA8, size, 1,
                                                     QRhiTexture::RenderTarget | QRhiTexture::UsedAsTransferSource));
-- 
2.51.2

